src/brain.py

```
@@ -74,34 +94,170 @@ async def _embed(text: str) -> np.ndarray | None:
+    if _is_weak_title(title) or _is_vague_description(description):
+        try:
+            from src.services.jina import fetch_markdown
+
+            jina_title, body = await fetch_markdown(url)
+            stronger = _clean_title(jina_title)
+            if stronger and not _is_weak_title(stronger):
+                title = stronger
+            paragraph = _first_paragraph(body)
+            if paragraph:
+                description = paragraph
+        except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)